### ISSUE TITLE
Analyzing source code to enforce the attributes

### ISSUE THREAD

**Issue by byme8** [protobuf-net/protobuf-net.Grpc#211]
I was playing around with the library and noticed that I constantly forget about contract attributes. it is not a big deal, but it can lead to some frustration when you are trying to figure out why it doesn't work.

In other libraries, I saw ways how to solve this issue. 

The solution may look like that:
https://cloud.githubusercontent.com/assets/46207/20078766/3ea54f14-a585-11e6-9873-b99cb5d9efe5.gif

Basically, there is a Roslyn analyzer that will check the source code for attributes and report issues at compile time.
I know how to implement it and I can create a pull request. Does the library need such functionality?

**Comment by mgravell** [comment]
See: protobuf-net.BuildTools

**Comment by byme8** [comment]
Oh, I see. Thanks.
